setup.py

```diff
@@ -25,6 +27,108 @@
 except ImportError:
     pass
 
+
+def getfiles(path):
+    '''Retrieve file list within a directory
+
+    :param path: directory path
+    :type path: string
+    :returns: file list
+    :rtype: list
+    '''
+    for root, dirs, files in os.walk(path):
+        file_list = [os.path.join(root, file) for file in files]
+    return(file_list)
+
+
+def isdebianbased():
+    '''Check if it is a Debian based system
+
+    :returns: True if debian based, false otherwise.
+    :rtype: bool
+    .. note:: This is defined by looking at the python path
+       Currently I have not find a better way to find the correct installation
+       path in such case.
+
+       From : https://wiki.debian.org/Python
+       distutils setup scripts install files in /usr/local/ not sys.prefix
+       (which is normally /usr/). This is because /usr/ is reserved for files
+       installed from Debian packages. Note that
+       /usr/local/lib/pythonX.Y/dist-packages is in sys.path so that modules
+       not installed from Debian packages can still be accessed by the system
+       Python. Tools like debhelper pass the --install-layout=deb option to
+       the setup script while building a Debian package so that its installs
+       files into /usr/ not /usr/local/.
+
+    '''
+    if([x for x in sys.path if 'dist-package' in x]):
+        return(True)
+    else:
+        return(False)
+
+
+def check_conf_file_location():
+    '''Retrieve data_files values for setuptools.setup()
+
+    :returns: data_files value that can be used by setuptools.setup()
+              like the following example.
+              [('etc/', ['redfish-client/etc/redfish-client.conf']),
+               ('share/redfish-client/templates',
+                ['redfish-client/templates/manager_info.template',
+                 'redfish-client/templates/bla.template'])]
+    :rtype: structure compatible with setuptools.setup()
+    '''
+    file_location = []
+    # Add conf file according to prefix
+    if(distutils.sysconfig.PREFIX == '/usr' and not isdebianbased()):
+        file_location.append(('/etc/',
+                              getfiles('redfish-client/etc')))
+    else:
+        file_location.append(('etc/',
+                              getfiles('redfish-client/etc')))
+    # Add template files
+    file_location.append(('share/redfish-client/templates',
+                          getfiles('redfish-client/templates')))
+    return(file_location)
```

Comment:
This is in setup.py so more or less the build and install program for python pkg.
Note : This looks a bit weird because setup() is using a strange structure based on list and tuples but this is not hardcoded.